assets/src/edit-story/components/panels/link.js

```diff
@@ -0,0 +1,166 @@
+const InputContainer = styled.div`
+  width: 100%;
+  color: ${({ theme }) => rgba(theme.colors.mg.v4, 0.55)};
+  font-family: ${({ theme }) => theme.fonts.body2.family};
+  font-size: ${({ theme }) => theme.fonts.body2.size};
+  border: 1px solid ${({ theme }) => theme.colors.fg.v3} !important;
+  border-radius: 4px !important;
+  box-sizing: border-box;
+`;
+
+const BrandIcon = styled.img`
+  width: 64px;
+  height: 64px;
+  border-radius: 50%;
+  background-color: ${({ theme }) => theme.colors.fg.v3};
+  border: none;
+  margin-right: 8px;
+`;
+
+function LinkPanel({ selectedElements, onSetProperties }) {
+  const link = getCommonValue(selectedElements, 'link') || DEFAULT_LINK;
+  const isFill = getCommonValue(selectedElements, 'isFill');
+  const [state, setState] = useState({ link });
+  useEffect(() => {
+    setState({ link });
+    // TODO(wassgha) Implement page parsing
+    if (link.url) {
+      populateMetadata(link.url);
+    }
+  }, [link, populateMetadata]);
+  const handleSubmit = (evt) => {
+    onSetProperties(state);
+    evt.preventDefault();
+  };
+  const canLink = !isFill;
+  const populateMetadata = useCallback(
+    debounce(300, async (/** url */) => {
+      // TODO(wassgha): Implement getting the page metadata
+    })
+  );
+  return (
+    <SimplePanel
+      name="link"
+      title={__('Link', 'web-stories')}
+      onSubmit={handleSubmit}
+    >
+      <Row>
+        <Note>
+          {__('Enter an address to apply a 1 or 2 tap link', 'web-stories')}
+        </Note>
+      </Row>
+
+      <Row>
+        <InputContainer>
+          <Input
+            type="text"
+            disabled={!canLink}
+            onChange={(evt) =>
+              setState({
+                ...state,
+                link: { ...state.link, url: evt.target.value },
+              })
+            }
+            onBlur={(evt) =>
+              evt.target.form.dispatchEvent(new window.Event('submit'))
+            }
+            placeholder={__('Web address', 'web-stories')}
+            value={state.link && state.link.url}
+            isMultiple={link === ''}
+            expand
+          />
+        </InputContainer>
+      </Row>
+
+      <Row>
+        <InputContainer>
+          <Input
+            type="text"
+            onChange={(evt) =>
+              setState({
+                ...state,
+                link: { ...state.link, desc: evt.target.value },
+              })
+            }
+            onBlur={(evt) =>
+              evt.target.form.dispatchEvent(new window.Event('submit'))
+            }
+            placeholder={__('Optional description', 'web-stories')}
+            value={state.link && state.link.desc}
+            isMultiple={link === ''}
+            expand
+          />
+        </InputContainer>
+      </Row>
+      {/** TODO(@wassgha): Replace with image upload component */}
+      <Row>
+        <BrandIcon src={state.link && state.link.image} />
```

Comment:
`state.link.?image`. Btw, property name should also be `icon` imho.

Reply:
@swissspidy doesn't look like we have optional chaining, is it easy to add?

Reply:
We do have it and use it in some areas already

Reply:
Yeah. I thought I used it already once. @wassgha what was the linter error?

Reply:
It worked! My editor was misconfigured